### PR TITLE
support test-student only license

### PR DESCRIPTION
**Pull request by smallst**
fix ENG-926
adding test-student-only license.

when a teacher create test-student and he has the licenses, auto create a test-only license and apply to the test-student.
when a teacher has test-student and then he has the licenses/ when old test-only licenses expires and the teacher still has active licenses (new) , we show a button in license page which helps teacher to get the free test-only license and auto apply to the test-student.

image
image

## Summary by CodeRabbit

- **New Features**
	- Introduced a method for obtaining test student licenses in the teacher dashboard.
	- Added functionality to categorize and manage test licenses for teachers.
	- Enhanced the user interface with additional localization strings for improved guidance.

- **Bug Fixes**
	- Improved handling of various license states (expired, pending, available) in the prepaid management system.

- **Documentation**
	- Updated localization messages to enhance user understanding of test licenses.

- **Style**
	- Minor CSS adjustments to accommodate new button styling in the teacher dashboard.

### CODE REVIEW

File: ozaria/site/components/teacher-dashboard/BaseTeacherLicenses/common/LicenseCard.vue [hunk 1]
```diff
@@ -190,6 +202,13 @@ export default {
   font-weight: 600;
 }
 
+.special {
+    font-size: 12px;
```
Comment (mrfinch) [inline ozaria/site/components/teacher-dashboard/BaseTeacherLicenses/common/LicenseCard.vue:206]: is indentation here more than .dates section 🤔

File: ozaria/site/components/teacher-dashboard/BaseTeacherLicenses/PageLicenses.vue [hunk 2]
```diff
@@ -42,6 +42,18 @@ export default {
...
         link: 'https://docs.google.com/presentation/d/1SfM5ZMjae8wm8HESHoXXO0wBnKJmJD53BgtG9XwVW9k/edit?usp=sharing'
       }
+    },
+    showGetTestLicense () {
+      const testStudentRelation = (me.get('related') || []).filter(related => related.relation === 'TestStudent')[0]
+      if (!testStudentRelation) {
+        return false
+      }
+      const testLicense = _.find(this.activeLicenses, (prepaid) => prepaid.properties.testStudentOnly)
+      if (testLicense) {
```
Comment (mrfinch) [inline ozaria/site/components/teacher-dashboard/BaseTeacherLicenses/PageLicenses.vue:52]: return !testLicense

File: ozaria/site/components/teacher-dashboard/BaseTeacherLicenses/PageLicenses.vue [hunk 3]
```diff
@@ -80,6 +92,14 @@ export default {
       >
         {{ $t("courses.get_enrollments") }}
       </primary-button>
+      <primary-button
+        v-if="showGetTestLicense"
+        class="get-test-license"
+        @click="$emit('getTestLicense', { teacherId } )"
```
Comment (mrfinch) [inline ozaria/site/components/teacher-dashboard/BaseTeacherLicenses/PageLicenses.vue:98]: why do we need teacherId, is it different than me.id in any case?

File: ozaria/site/components/teacher-dashboard/BaseTeacherLicenses/common/LicenseCard.vue (hunk 1 above)
Comment (smallst) [reply ozaria/site/components/teacher-dashboard/BaseTeacherLicenses/common/LicenseCard.vue:206]: ha cool. eslint doesn't fix style indentation.

File: ozaria/site/components/teacher-dashboard/BaseTeacherLicenses/PageLicenses.vue (hunk 3 above)
Comment (smallst) [reply ozaria/site/components/teacher-dashboard/BaseTeacherLicenses/PageLicenses.vue:98]: i'm not sure, this `teacherId` is from vuex store and having very complicated cases so I cannot say it always to be me.id lol

Comment (smallst) [reply ozaria/site/components/teacher-dashboard/BaseTeacherLicenses/PageLicenses.vue:98]: maybe school admin can see teachers' licenses without spy them ? so me.id is different?

**Pull request merged**